### PR TITLE
Fix 3.3.1 deps

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

`conda-forge::matplotlib-base=3.3.1` is missing the newly added direct dependency on `certifi`.
This causes other Python packages (e.g., `pip`/`pkg_resources`) to report unsatisfied dependencies in case an older `certifi` is installed.

Other dependencies in the recipe did not have version constrains, which is also addressed in this PR.

cc @dlaehnemann

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -41,14 +40,15 @@ outputs:
         - python
         - setuptools
         - {{ pin_compatible('numpy') }}
+        - certifi >=2020.06.20
```
Comment (mbargull) [inline recipe/meta.yaml:43]: ref: https://github.com/matplotlib/matplotlib/pull/18185/files#diff-2eeaed663bd0d25b7e608891384b7298R281

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,8 +28,7 @@ outputs:
       host:
         - python
         - pip
-        - numpy 1.15  # [x86_64 and python_impl=="cpython" and py<38]
-        - numpy 1.16  # [x86_64 and python_impl=="cpython" and py==38]
+        - numpy 1.16  # [x86_64 and python_impl=="cpython"]
```
Comment (mbargull) [inline recipe/meta.yaml:31]: ref: https://github.com/matplotlib/matplotlib/pull/17662/files#diff-2eeaed663bd0d25b7e608891384b7298L283-R286

File: recipe/meta.yaml [hunk 3]
```diff
@@ -41,14 +40,15 @@ outputs:
...
         - setuptools
         - {{ pin_compatible('numpy') }}
+        - certifi >=2020.06.20
         - cycler >=0.10
-        - python-dateutil
+        - python-dateutil >=2.7
         - pillow >=6.2.0
         - freetype
-        - pyparsing
+        - pyparsing >=2.2.1
```
Comment (mbargull) [inline recipe/meta.yaml:48]: ref: https://github.com/matplotlib/matplotlib/pull/17662/files#diff-2eeaed663bd0d25b7e608891384b7298L283-R286

**Comment by dopplershift** [comment]
So the certifi update makes sense, but the rest of those versions you've pulled from the master branch which will become 3.4. For 3.3x, you need to look at the appropriate branch which shows:
https://github.com/matplotlib/matplotlib/blob/355fac9d825883931d42e466e8797bf6b10b3f07/setup.py#L276-L289

**Comment by mbargull** [comment]
> So the certifi update makes sense, but the rest of those versions you've pulled from the master branch which will become 3.4.

Ah, my bad. I had both https://github.com/matplotlib/matplotlib/blob/v3.3.1/setup.py and https://github.com/matplotlib/matplotlib/blob/master/setup.py open and ended up looking at the wrong one...

Fixed now. Thanks for catching my mistake.

**Pull request merged**